master/buildbot/db/buildrequests.py

```diff
@@ -220,34 +227,67 @@ def getResults(query):
         return self.db.pool.do(thd)
 
     @with_master_objectid
-    def getOldestBuildRequestInQueue(self, buildername, _master_objectid=None):
+    def getPrioritizedBuildRequestsInQueue(self, buildername, queue=None, _master_objectid=None):
```

Comment:
Can you write some docs as to what each of the parameters do here? Specifically queue